fairlearn/reductions/_grid_search/grid_search.py

```diff
@@ -179,21 +177,17 @@ def fit(self, X, y, **kwargs):
             feature used by the constraints object
         :type sensitive_features: numpy.ndarray, pandas.DataFrame, pandas.Series, or list (for now)
         """
-        X_train, y_train, _ = _validate_and_reformat_reductions_input(
-            X, y, enforce_binary_sensitive_feature=True, **kwargs)
-
         if isinstance(self.constraints, ClassificationMoment):
             logger.debug("Classification problem detected")
             is_classification_reduction = True
-            # We have a classification problem
-            # Need to make sure that y is binary (for now)
-            unique_labels = np.unique(y_train)
-            if not set(unique_labels).issubset({0, 1}):
-                raise RuntimeError(self._MESSAGE_Y_NOT_BINARY)
         else:
             logger.debug("Regression problem detected")
```

Comment:
Don't regression problems require an instance of `ConditionalLossMoment`?

Reply:
The logic should be identical. I didn't change anything about moments here AFAICT

Reply:
Fair point.